src/database/models/image_item.rs

```diff
@@ -0,0 +1,242 @@
+use super::ids::*;
+use crate::database::models::DatabaseError;
+use chrono::{DateTime, Utc};
+use redis::cmd;
+use serde::{Deserialize, Serialize};
+
+const IMAGES_NAMESPACE: &str = "images";
+const DEFAULT_EXPIRY: i64 = 1800; // 30 minutes
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Image {
+    pub id: ImageId,
+    pub url: String,
+    pub size: u64,
+    pub created: DateTime<Utc>,
+    pub owner_id: UserId,
+
+    // context it is associated with
+    pub context_type_id: ImageContextTypeId,
+    pub context_id: Option<u64>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct QueryImage {
+    pub id: ImageId,
+    pub url: String,
+    pub size: u64,
+    pub created: DateTime<Utc>,
+    pub owner_id: UserId,
+
+    // context it is associated with
+    pub context_type_name: String,
+    pub context_id: Option<i64>,
+}
```

Comment:
don't think these need to be separate types. we only have them as seperate on projects/versions/files for legacy reasons. They seem mostly identical